compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -155,6 +155,11 @@ impl Binary {
                 }
             }
             BinaryOp::Sub { .. } => {
+                if dfg.resolve(self.lhs) == dfg.resolve(self.rhs) {
```

Comment:
```suggestion
                if lhs == rhs {
```
The rest of this function has been updated since the PR was opened to resolve these beforehand